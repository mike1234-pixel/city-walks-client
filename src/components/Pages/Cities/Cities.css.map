{"version":3,"sourceRoot":"","sources":["Cities.scss"],"names":[],"mappings":"AAAA;EACE;AACA;EACA;AACA;EACA;;;AAGF;EACE;;;AAGF;EACE","file":"Cities.css"}