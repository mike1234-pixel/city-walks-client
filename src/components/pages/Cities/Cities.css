.city-card {
  transition: transform 0.2s;
  background: var(--dark);
  color: white;
}

.view {
  background: var(--linear-blue);
}

.city-card:hover {
  transform: scale(1.05);
}
