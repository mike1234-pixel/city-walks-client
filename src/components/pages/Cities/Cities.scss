.city-card {
  transition: transform 0.2s; /* Animation */
  background: var(--elegant-color-dark);
  /* background: blue; */
  color: white;
}

.view {
  background-color: var(--blue);
}

.city-card:hover {
  transform: scale(1.05);
}
