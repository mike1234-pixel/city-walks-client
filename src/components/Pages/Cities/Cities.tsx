import { useEffect, useState } from "react";
import { MDBContainer } from "mdbreact";
import ReactPaginate from "react-paginate";
import { connect } from "react-redux";
import { bindActionCreators, Dispatch } from "redux";
import { handleClickSearch, setRedirect } from "../../../actions/actions";
import City from "../../../types/Cities/City";
import { RootState } from "../../../store";
import CityCard from "./CityCard";
import * as Actions from "../../../types/Actions";
import "./Cities.css";

interface CitiesProps {
  cities: Array<City>;
  handleClickSearch: (cityToSearch: string) => Actions.HandleClickSearch;
  setRedirect: (redirect: boolean) => Actions.SetRedirect;
}

const Cities = (props: CitiesProps) => {

  const { cities, handleClickSearch, setRedirect } = props;

  const [pageNumber, setPageNumber] = useState<number>(0);

  const citiesPerPage = 3;
  const pagesVisited = pageNumber * citiesPerPage;

  const pageCount = Math.ceil(cities.length / citiesPerPage);

  const handleSubmitSearch: (cityName: string) => void = (cityName) => {
    handleClickSearch(cityName);
    setRedirect(true);
  };

  const handlePageChange = ({ selected }: { selected: number; }) => {
    setPageNumber(selected);
  };

  useEffect(() => {
    window.scrollTo(0, 0);
  }, []);

  return (
    <MDBContainer>
      <div className='cities-page-container page'>
        <div
          data-testid='cities-page-heading'
          className='page-heading-container'
        >
          <h1 className='page-heading'>Cities</h1>
          <h2 className='page-subheading'>search walks by city</h2>
        </div>
        <div className='card-container'>
          {cities
            .slice(pagesVisited, pagesVisited + citiesPerPage)
            .map((city) => {
              const { _id, city: cityName, description, img } = city;

              return (
                <div key={_id}>
                  <CityCard
                    id={_id}
                    name={cityName}
                    description={description}
                    imgSrc={img}
                    handleSubmitSearch={handleSubmitSearch}
                  />
                </div>
              );
            })}
        </div>
        <ReactPaginate
          previousLabel={"Previous"}
          nextLabel={"Next"}
          pageCount={pageCount}
          onPageChange={handlePageChange}
          containerClassName={"pagination-buttons"}
          previousLinkClassName={"previous-button"}
          nextLinkClassName={"next-button"}
          disabledClassName={"pagination-disabled"}
          activeClassName={"pagination-active"}
        />
      </div>
    </MDBContainer>
  );
};

const mapStateToProps: (state: RootState) => void = (state) => ({
  cities: state.citiesState.cities,
});

const mapDispatchToProps = (dispatch: Dispatch) => {
  return {
    dispatch,
    ...bindActionCreators({ handleClickSearch, setRedirect }, dispatch),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(Cities);
