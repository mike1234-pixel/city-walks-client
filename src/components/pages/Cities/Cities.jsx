import { useContext, useEffect } from "react"
import { SearchContext } from '../../../context/SearchContext'
import { MDBIcon, MDBBtn, MDBCard, MDBCardBody, MDBCardImage, MDBCardTitle, MDBCardText, MDBCol } from 'mdbreact'
import PopUp from "../../PopUp/PopUp"
import { LoginContext } from "../../../context/LoginContext"
import './Cities.css'

const Cities = (props) => {

    const data = props.cities

    const { handleClickSearch } = useContext(SearchContext)
    const { popupVisible } = useContext(LoginContext)

    useEffect(() => {
        window.scrollTo(0, 0);
      }, []);

    const cities = data.map((city) => {
        return (
        <MDBCol key={city._id}>
        <MDBCard className="city-card">
            <MDBCardImage className="cutter img-fluid" src={city.img} alt={city.city} waves/>
            <MDBCardBody>
            <MDBCardTitle>{city.city}</MDBCardTitle>
            <MDBCardText>{city.description}</MDBCardText>
            <MDBBtn outline color="white" className="city-card-btn" onClick={() => handleClickSearch(city.city)}>Find walks <MDBIcon icon="search"/></MDBBtn>
            </MDBCardBody>
        </MDBCard>
        </MDBCol>
        )
    })

    return (
        <div className="cities-page-container min-page-height">
            {popupVisible && <PopUp/>}
            <div data-testid="cities-page-heading" className="page-heading-container">
                <h1 className="page-heading">Cities</h1>
            </div>
                <div className="card-container">{cities}</div>
        </div>
    )
}

export default Cities

// these cards will conduct a search and filter the walks by city -- each one will link to the walks page