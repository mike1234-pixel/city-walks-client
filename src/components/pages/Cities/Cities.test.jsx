import renderer from 'react-test-renderer'
import { SearchContextProvider } from "../../../context/SearchContext"
import { LoginContextProvider } from "../../../context/LoginContext"
import CitiesTestData from "../../../container/CitiesTestData.json"
import Cities from './Cities'

// snapshot test
it('Cities Page renders correctly', () => {  

  const cities = CitiesTestData 

  const tree = renderer
    .create(
    <LoginContextProvider>
      <SearchContextProvider>
        <Cities cities={cities}/>
      </SearchContextProvider>
    </LoginContextProvider>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});