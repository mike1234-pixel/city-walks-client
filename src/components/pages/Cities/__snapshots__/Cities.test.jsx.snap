// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cities Page renders correctly 1`] = `
<div
  className="cities-page-container min-page-height"
>
  <div
    className="page-heading-container"
    data-testid="cities-page-heading"
  >
    <h1
      className="page-heading"
    >
      Cities
    </h1>
  </div>
  <div
    className="card-container"
  >
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Liverpool"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.unsplash.com/photo-1505833779582-e28ba8f922f8…VufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1049&q=80"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Liverpool
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            A key trade and migration port from the 18th to the early 20th centuries, it's also, famously, the hometown of The Beatles.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Edinburgh"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.unsplash.com/photo-1506377585622-bedcbb027afc…VufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1050&q=80"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Edinburgh
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Edinburgh is Scotland's compact, hilly capital. It has a medieval Old Town and elegant Georgian New Town with gardens.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Budapest"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.unsplash.com/photo-1565426873118-a17ed65d74b9…VufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1055&q=80"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Budapest
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Budapest, Hungary’s capital, is bisected by the River Danube. Its 19th-century Chain Bridge has visitors all year round.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="London"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/460672/pexels-photo-460672.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            London
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            London, the capital of England and the United Kingdom, is a 21st-century city with history stretching back to Roman times.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card city-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Amsterdam"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/1329510/pexels-photo-1329510.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            Amsterdam
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Amsterdam is the Netherlands’ capital, known for its artistic heritage, elaborate canal system and narrow houses with gabled facades.
          </p>
          <button
            className="btn-white btn-outline-white btn Ripple-parent city-card-btn"
            data-test="button"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
            type="button"
          >
            Find walks 
            <i
              className="fa fa-search"
              data-test="fa"
            />
            <div
              className="Ripple Ripple-outline "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
